src/extension.ts

```diff
@@ -93,12 +93,18 @@ function initRPC(clientID: string): void {
 		setActivity();
 		// Set the activity once on ready
 		setTimeout(() => rpc.setActivity(activity), 500);
-		eventHandlers.add(workspace.onDidChangeTextDocument(() => setActivity()))
-			.add(workspace.onDidOpenTextDocument(() => setActivity()))
-			.add(workspace.onDidCloseTextDocument(() => setActivity()))
-			.add(debug.onDidChangeActiveDebugSession(() => setActivity()))
+		eventHandlers.add(debug.onDidChangeActiveDebugSession(() => setActivity()))
 			.add(debug.onDidStartDebugSession(() => setActivity()))
 			.add(debug.onDidTerminateDebugSession(() => setActivity()));
+		if (config.get('workspaceElapsedTime')) {
```

Comment:
I think this can be done nicer tbh. Instead of splitting up the handlers and having duplicate code:

```js
const workspaceElapsedTime = Boolean(config.get('workspaceElapsedTime'));

eventHandlers.add(workspace.onDidChangeTextDocument(() => setActivity(true)))
  .add(workspace.onDidOpenTextDocument(() => setActivity(workspaceElapsedTime)))
  .add(workspace.onDidCloseTextDocument(() => setActivity(workspaceElapsedTime)));
```

Reply:
I kind of felt I wasn't doing enough for this bit as well - I'll get on it.